app/src/main/java/com/github/wnder/GuessLocationActivity.java

```diff
@@ -166,108 +218,119 @@ public boolean onMapClick(@NonNull LatLng point) {
+        if(compassMode) updateCompassMode(); //Would be nice to find a way to know if the map is rotating instead of simply moving
     }
 
     /**
-     * resume mapbox
+     *  Update the compass and the hotbar according to the mode and distance between the current guess and picture location
      */
-    @Override
-    protected void onResume() {
-        super.onResume();
-        mapView.onResume();
-    }
+    private void updateCompassMode(){
+        SymbolLayer layer = (SymbolLayer) mapboxMap.getStyle().getLayer(ARROW_LAYER_ID);
+        View hotbarView = findViewById(R.id.hotbarView);
 
-    /**
-     * pause mapbox
-     */
-    @Override
-    protected void onPause() {
-        super.onPause();
-        mapView.onPause();
-    }
+        if (!compassMode) {
+            layer.setProperties(PropertyFactory.visibility(Property.NONE));
+            hotbarView.setVisibility(View.INVISIBLE);
 
-    /**
-     * stop mapbox
-     */
-    @Override
-    protected void onStop() {
-        super.onStop();
-        mapView.onStop();
+        } else {
+            if (!guessIsClose()) { //compass update
+                double mapRotation = mapboxMap.getCameraPosition().bearing; //rotation of the map
+                float pictureRotation = 0; //TODO rotation of the picture location from the current position
+                layer.setProperties(PropertyFactory.iconRotate((float) -mapRotation));
+
+                layer.setProperties(PropertyFactory.visibility(Property.VISIBLE));
+                hotbarView.setVisibility(View.INVISIBLE);
+
+            } else { //hotbar update
+                //TODO update hotbar
+
+                layer.setProperties(PropertyFactory.visibility(Property.NONE));
+                hotbarView.setVisibility(View.VISIBLE);
+            }
+        }
     }
 
     /**
-     * save mapbox instance state
-     * @param outState output
+     * Return true if the guess is close enough to display the hotbar instead of the compass
      */
-    @Override
-    protected void onSaveInstanceState(Bundle outState) {
-        super.onSaveInstanceState(outState);
-        mapView.onSaveInstanceState(outState);
+    private boolean guessIsClose() {
+        return false; //TODO compute or hardcode a distance in which the hotbar should be displayed
     }
 
     /**
-     * when memory is low
+     * Switch between compass mode and normal mode
      */
-    @Override
-    public void onLowMemory() {
-        super.onLowMemory();
-        mapView.onLowMemory();
+    private void switchMode() {
+        //If the guess has already been done or that the map didn't load yet do not switch mode
+        if (guessConfirmed || mapboxMap.getStyle() == null) {
+            return;
+        }
+
+        TextView compassModeButtonView = (TextView) findViewById(R.id.compassMode);
+
+        compassMode = !compassMode;
+        updateCompassMode();
+        if(compassMode) {
+            updateGuessPositionFromGPS.run();
```

Comment:
I'm not sure how this works exactly but I'm afraid this will run a new timer multiple time if you enable/disable compass mode multiple time

Reply:
It shoudn't since the timer start at the beginning of the class and run behind the scene and that if you're in Compass Mode or not. The timer will just not trigger the updateGuessPositionFromGPS function every 10 seconds if you're in Nomral Mode since <ou don't need to.
But I start the timer only once at the start of the activity and not when it switches mode

Reply:
Ookok I misunderstood the use of .run() here.